internal/app/machined/pkg/controllers/k8s/static_pod_server.go

```diff
@@ -136,7 +136,7 @@ func (ctrl *StaticPodServerController) createServer(ctx context.Context, r contr
 		}
 	})
 
-	listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
this is good, but I guess the real issue is that it doesn't correctly restart the server if it fails to listen.